### PR TITLE
Support AIX for get_peer_cred

### PR DESCRIPTION
**Pull request by ecnelises**
## Motivation
We're proposing AIX on PowerPC as Tier-3 target (see https://github.com/rust-lang/compiler-team/issues/553). And we encountered build failure of tokio for AIX.

## Solution
Implement AIX's version of `get_peer_cred` which is almost the same as FreeBSD's. Since AIX hasn't been officially accepted, this is in draft status.

### CODE REVIEW

**Comment by Noah-Kennedy** [comment]
Is this in stable yet?

**Comment by Darksonn** [comment]
Can you please clarify the state of this PR?

**Comment by ecnelises** [comment]
Yes, AIX recognition has been in Rust stable (which means no warnings will be emitted), although some dependent crates are not available yet. This patch has left draft status and is ready to be checked-in after approval.

**Comment by Darksonn** [comment]
Is it possible to add this platform to our CI setup? We cross-compile to other weird targets in the following CI check:

https://github.com/tokio-rs/tokio/blob/e629ad7c9ac48df0cede0620c44e0ea652f45af9/.github/workflows/ci.yml#L256-L278

The `cross-test` check also runs tests on the platform, but that's not possible for all platforms.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -261,6 +261,7 @@ jobs:
         target:
           - powerpc-unknown-linux-gnu
           - powerpc64-unknown-linux-gnu
+          - powerpc64-ibm-aix
```
Comment (Darksonn) [inline .github/workflows/ci.yml:264]: Unfortunately, this fails with the following error:
```
$ rustup toolchain install stable --target powerpc64-ibm-aix --profile minimal --no-self-update
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: latest update on 2023-02-09, rust version 1.67.1 (d5a82bbd2 2023-02-07)
error: component 'rust-std' for target 'powerpc64-ibm-aix' is unavailable for download for channel 'stable'
If you don't need the component, you could try a minimal installation with:

    rustup toolchain add stable --profile minimal
```
Perhaps adding this to CI is not possible today?

Comment (ecnelises) [reply .github/workflows/ci.yml]: Ah, I forgot about it. I'll add it back once `powerpc64-ibm-aix` is available in rustup.

**Review by Darksonn — APPROVED** [review]
Since this cannot be tested on CI today, I am ok with merging it as-is. I would appreciate a CI PR if AIX resources become available in rustup in the future.

Have you executed this code at least once to sanity check it? If so, then I am happy with merging it as-is.

**Comment by Darksonn** [comment]
Ping regarding my question on whether you have executed this?

**Comment by ecnelises** [comment]
Hi, sorry for the very late response, this builds successfully on AIX, and the function works as expected on AIX. (I cannot run the whole test suite for tokio now because of some issues met in mio)

**Pull request merged**